sdlib/d/gc/extent.d

```diff
@@ -217,13 +280,22 @@ public:
 		return (bits >> 48) & Mask;
 	}
 
-	uint allocate() {
+	uint allocate(bool isAppendable = false, bool isFinalizable = false) {
```

Comment:
This is unlikely to be the API we want. We are going to allocate slots in bulk and store them in a per thread cache, as per #269 so this API already doesn't work, unless we want to have 4 caches for every combination of flags, which doesn't seems workable.

Reply:
Why would it be necessary to have separate caches for the combinations of flags when caching per thread? The flags are only set when a slot is occupied.

Also, I thought we were going to start with the simplest mechanisms and then refine?

Reply:
@deadalnix do you have a description of the intended cache scheme somewhere that would explain why using this scheme would require a 4x larger cache? Slabs were not even mentioned in https://github.com/snazzy-d/sdc/issues/269 or the linked snippets from jemalloc.

Reply:
@deadalnix Finally understood what you meant here, following our Slack conversation re: when in the pipeline the metadata ought to be set (i.e. in `tcache` rather than in `Extent` or the arena.)